benchmark/benchmark.py

```diff
@@ -0,0 +1,223 @@
+        # TODO: Review if we should use sync or psync. We use `sync` in other benchmarks.
+        "IO_ENGINE": str("libaio" if cfg['direct_io'] else "psync"),
+    }
+    log.info(f"Running FIO with args: %s; env: %s", subprocess_args, subprocess_env)
+
+    # Use Popen instead of check_output, as we had some issues when trying to attach perf
+    with Popen(subprocess_args, env=subprocess_env) as process:
+        exit_code = process.wait()
+        if exit_code != 0:
+            log.error(f"FIO process failed with exit code {exit_code}")
+            raise subprocess.CalledProcessError(exit_code, subprocess_args)
+        else:
+            log.info("FIO process completed successfully")
+
+
+def _unmount_mp(mount_dir: str) -> None:
+    """
+    Attempts to unmount Mountpoint
+    """
+    subprocess.check_output(["/usr/bin/umount", mount_dir])
+    log.info(f"{mount_dir} unmounted")
+
+
+def _collect_logs() -> None:
+    """
+    Collect the Mountpoint log if it exists and move to the output directory.
+    Mountpoint log filename will be normalized removing the date, etc..
+    The old log directory is removed.
+
+    Fails if more than one log file is found.
+    """
+    logs_directory = path.join(os.getcwd(), MP_LOGS_DIRECTORY)
+    dir_entries = os.listdir(logs_directory)
+
+    if not dir_entries:
+        log.debug(f"No Mountpoint log files in directory {logs_directory}")
+        return
+
+    assert len(dir_entries) <= 1, f"Expected no more than one log file in {logs_directory}"
+
+    old_log_dir = path.join(logs_directory, dir_entries[0])
+    new_log_path = "mountpoint-s3.log"
+    log.debug(f"Renaming {old_log_dir} to {new_log_path}")
+    os.rename(old_log_dir, new_log_path)
+    os.rmdir(logs_directory)
+
+
+def _write_metadata(metadata: dict[str, any]) -> None:
+    with open("metadata.json", "w") as f:
+        json_str = json.dumps(metadata, default=str)
+        f.write(json_str)
+
+
+def _postprocessing(metadata: dict[str, any]) -> None:
+    _collect_logs()
+    _write_metadata(metadata)
+
+
+@hydra.main(version_base=None, config_path="conf", config_name="config")
+def run_experiment(cfg: DictConfig) -> None:
+    """
+    At a high level, we want to mount the S3 bucket using Mountpoint,
+    run a synthetic workload against Mountpoint while capturing metrics and logs,
+    then end the load and unmount the bucket.
+
+    We should collect all of the logs and metric and dump them in the output directory.
+    """
+    log.info("Experiment starting")
+    success = False
+    mounted = False
+    start_time = datetime.now(tz=timezone.utc)
+    metadata = {
+        "start_time": start_time,
+    }
+
+    try:
+        mount_dir = tempfile.mkdtemp(suffix=".mountpoint-s3")
+        mp_version = _mount_mp(cfg, metadata, mount_dir)
+        mounted = True
+        metadata["mp_version"] = mp_version
+    except subprocess.SubprocessError as e:
+        log.error(f"Error during mounting: {e}")
```

Comment:
`try/except/else` appears to be what we want here, and could mean we drop the `mounted` variable

Reply:
Though in production quality code, we'd probably want this to be a `with` block